pkg/backend/httpstate/backend.go

```diff
@@ -158,9 +158,14 @@ func New(d diag.Sink, cloudURL string, project *workspace.Project, insecure bool
 }
 
 // loginWithBrowser uses a web-browser to log into the cloud and returns the cloud backend for it.
-func loginWithBrowser(ctx context.Context, d diag.Sink, cloudURL string, project *workspace.Project,
-	insecure bool, opts display.Options,
-) (Backend, error) {
+func loginWithBrowser(
+	ctx context.Context,
+	cloudURL string,
+	insecure bool,
+	welcome func(display.Options),
+	current bool,
+	opts display.Options,
```

Comment:
opts looks to only be used to pass to welcome? Just make welcome a func() and don't pass opts instead?

Reply:
ah, nice--yes, we can do that.

Reply:
oh wait, no--opts is also used for colorizing the various messages we print out.